model-optimizer/extensions/front/tf/CTCGreedyDecoderReplacement.py

```diff
@@ -16,89 +16,60 @@
                    ],
             edges=[('decoder', 'sparse_to_dense', {'out': 0}),
+                   ('decoder', 'sparse_to_dense', {'out': 2}),
                    ('decoder', 'cast', {'out': 1}),
                    ('cast', 'sparse_to_dense', {'out': 0})
                    ]
         )
 
     def replace_sub_graph(self, graph: Graph, match: dict):
         # TODO: Once Inference Engine's CTCGreedyDecoder starts to support sequence length format like in TensorFlow,
-        # CTCGreedyDecoderReplacement2 needs to be removed and CTCGreedyDecoderReplacement, a more generic
-        # transformation, needs to be adopted for all cases
-        ctc_greedy_decoder = match['decoder']
+        ctc_greedy_decoder_tf = match['decoder']
         cast = match['cast']
         sparse_to_dense = match['sparse_to_dense']
-        sparse_to_dense_name = sparse_to_dense.soft_get('name', sparse_to_dense.id)
-
-        # disconnect SparseToDense and Cast nodes
-        sparse_to_dense.in_port(0).disconnect()
-        cast.in_port(0).disconnect()
-
-        # transform CTCGreedyDecoder output to TensorFlow's one:
-        # 1. squeeze the output to [N, T] shape
-        # 2. cast it to integer
-        squeeze_dec_seq = create_op_with_const_inputs(graph, Squeeze, {1: int64_array([2, 3])},
-                                                      {'name': sparse_to_dense_name})
-        squeeze_dec_seq.in_port(0).connect(ctc_greedy_decoder.out_port(0))
-        cast_to_int = Cast(graph, {'name': sparse_to_dense_name + '/CastToInt',
-                                   'dst_type': np.int32}).create_node()
-        cast_to_int.in_port(0).connect(squeeze_dec_seq.out_port(0))
-
-        # preserve output name from original graph
-        rename_nodes([(sparse_to_dense, sparse_to_dense_name + '/AbandonedName'),
-                      (cast_to_int, sparse_to_dense_name)])
 
+        # for normalizing input chanel need to transpose input data from [T, N, C] to [N, T, C]
+        # which supported CTCGreedyDecoderSeqLen op.
+        ctc_data_permute = create_op_with_const_inputs(graph, Transpose, {1: int64_array([1, 0, 2])},
+                                                            {'name': ctc_greedy_decoder_tf.name + '/ctc_data_permute'})
+        ctc_greedy_decoder_tf.in_port(0).get_source().connect(ctc_data_permute.in_port(0))
+        merge_repeated_tf = ctc_greedy_decoder_tf.soft_get('merge_repeated', ctc_greedy_decoder_tf.id)
+        ctc_greedy_decoder = CTCGreedyDecoderSeqLenOp(graph, {'name': ctc_greedy_decoder_tf.name,
+                                                              'cmerge_repeated': merge_repeated_tf}).create_node()
+        ctc_greedy_decoder.in_port(0).connect(ctc_data_permute.out_port(0))
+        ctc_greedy_decoder_tf.in_port(1).get_source().connect(ctc_greedy_decoder.in_port(1))
+        ctc_res1 = Result(graph, {'name': ctc_greedy_decoder_tf.soft_get('name', ctc_greedy_decoder_tf.id) + '/Result'}).create_node()
```

Comment:
Is it suggested by @lazarevevgeny? I think it is good to have auxiliary result node.
nit: please add a comment for this transformation about why additional `Result` node. This is because of different output format TF vs. OV.